bees/discordbee/discordbeefactory.go

```diff
@@ -0,0 +1,151 @@
+/*
+ *    Copyright (C) 2019      CalmBit
+ *                  2014-2019 Christian Muehlhaeuser
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU Affero General Public License as published
+ *    by the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Authors:
+ *      CalmBit <calmbit@posto.net>
+ *      Christian Muehlhaeuser <muesli@gmail.com>
+ */
+
+package discordbee
+
+import (
+	"github.com/muesli/beehive/bees"
+)
+
+// DiscordBeeFactory is a factory for DiscordBees.
+type DiscordBeeFactory struct {
+	bees.BeeFactory
+}
+
+// New returns a new Bee instance configured with the supplied options.
+func (factory *DiscordBeeFactory) New(name, description string, options bees.BeeOptions) bees.BeeInterface {
+	bee := DiscordBee{
+		Bee: bees.NewBee(name, factory.ID(), description, options),
+	}
+	bee.ReloadOptions(options)
+
+	return &bee
+}
+
+// ID returns the ID of this Bee.
+func (factory *DiscordBeeFactory) ID() string {
+	return "discordbee"
+}
+
+// Name returns the name of this Bee.
+func (factory *DiscordBeeFactory) Name() string {
+	return "Discord"
+}
+
+// Description returns the description of this Bee.
+func (factory *DiscordBeeFactory) Description() string {
+	return "Connects to Discord as a bot!"
+}
+
+// Image returns the filename of an image for this Bee.
+func (factory *DiscordBeeFactory) Image() string {
+	return factory.ID() + ".png"
+}
+
+// LogoColor returns the preferred logo background color (used by the admin interface).
+func (factory *DiscordBeeFactory) LogoColor() string {
+	return "#7289DA"
+}
+
+// Options returns the options available to configure this Bee.
+func (factory *DiscordBeeFactory) Options() []bees.BeeOptionDescriptor {
+	opts := []bees.BeeOptionDescriptor{
+		{
+			Name:        "api_token",
+			Description: "The Discord API token for your bot",
+			Type:        "string",
+			Mandatory:   true,
+		},
+		{
+			Name:        "trigger_phrase",
+			Description: "The phrase/word to start a command with in order for Beehive to recognize it",
+			Type:        "string",
+			Mandatory:   true,
+		},
+	}
+	return opts
+}
+
+// Events describes the available events provided by this Bee.
+func (factory *DiscordBeeFactory) Events() []bees.EventDescriptor {
+	events := []bees.EventDescriptor{
+		{
+			Namespace:   factory.Name(),
+			Name:        "message_received",
```

Comment:
Unless you want to re-use the "receive" or "send" keywords in events & actions (for receiving/sending some other than messages), I would suggest simply calling this event `message` and the action `send`.